Mathlib/Tactic/TFAE.lean

```diff
@@ -213,8 +219,12 @@ elab_rules : tactic
     let Pi := tfaeList.get! (i'-1)
     let Pj := tfaeList.get! (j'-1)
     let type ← mkImplType Pi arr Pj
-    let (goal1, goal2) ← tfaeHaveCore goal h i j arr type
-    replaceMainGoal [goal1, goal2]
+    let proof? ← pf.mapM fun pf => Tactic.elabTermEnsuringType pf type
+    let (arrowGoal?, mainGoal) ← tfaeHaveCore goal h i j arr type proof?
+    if let some arrowGoal := arrowGoal? then
+      replaceMainGoal [arrowGoal, mainGoal]
+    else
+      replaceMainGoal [mainGoal]
```

Comment:
```suggestion
     replaceMainGoal [arrowGoal?, some mainGoal].reduceOption
```
if you like it!

Reply:
That's definitely golfed but let me see if we want to push the faithful `have` support commit first, as that would obliterate this! :)